src/App.jsx

```diff
@@ -1,14 +1,65 @@
 import './App.css';
+import ChatLog from './components/ChatLog';
+import ColorChoice from './components/ColorChoice';
+import DATA from './data/messages.json';
+import { useState } from 'react';
 
 const App = () => {
+  const [entryData, setEntryData] = useState(DATA);
+  const [localColor, setLocalColor] = useState('green');
+  const [remoteColor, setRemoteColor] = useState('blue');
+
+  const LOCAL_SENDER = entryData[0].sender;
+  const REMOTE_SENDER = entryData[1].sender;
+
+  const handleColorChange = (sender, color) => {
+    if (sender === LOCAL_SENDER){
+      setLocalColor(color);
+    } else if (sender === REMOTE_SENDER) {
+      setRemoteColor(color);
+    }
+  };
+
+  const updateEntryLikedState = (entryId) => {
+    setEntryData(entries => {
+      return entries.map(entry => {
+        if (entry.id === entryId) {
+          return { ...entry, liked: !entry.liked };
+        } else {
+          return entry;
+        }
+      });
+    });
+  };
+
+  const totalLikes = entryData.reduce((sum, entry) => {
+    return entry.liked ? sum + 1 : sum;
```

Comment:
Nice job using `reduce`! This is how most JS devs would go about calculating a total amount too. 

Instead of using a ternary operator here, we could write something like:
```suggestion
    (totalLikes + message.liked)
```

Adding a boolean to a number treats true as 1 and 0 as false